cmd/clusterctl/client/config/imagemeta_client.go

```diff
@@ -119,53 +118,20 @@ func (i *imageMeta) Union(other *imageMeta) {
 
 // ApplyToImage changes an image name applying the transformations defined in the current imageMeta.
 func (i *imageMeta) ApplyToImage(image string) (string, error) {
```

Comment:
cc @fabriziopandini @wfernandes 

to make sure we have tests and the logic looks good

Reply:
In `imagemeta_client_test.go` there were no specific unit tests for `ApplyToImage` but there were tests for `AlterImage` which calls out to `ApplyImage`. That being said I see that a lot of tests were added for `ImageFromString`. 👍